Sources/AsyncHTTPClient/ConnectionPool/HTTPConnectionPool+Waiter.swift

```diff
@@ -0,0 +1,64 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the AsyncHTTPClient open source project
+//
+// Copyright (c) 2021 Apple Inc. and the AsyncHTTPClient project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of AsyncHTTPClient project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIO
+
+extension HTTPConnectionPool {
+    struct RequestID: Hashable {
+        private let objectIdentifier: ObjectIdentifier
+
+        init(_ request: HTTPScheduledRequest) {
+            self.objectIdentifier = ObjectIdentifier(request)
+        }
+    }
+
+    struct Waiter {
+        var requestID: RequestID {
+            RequestID(self.request)
+        }
+
+        var request: HTTPScheduledRequest {
+            didSet {
+                self.updateEventLoopRequirement()
+            }
+        }
```

Comment:
@glbrntt Change to `RecordID`. But I moved the `RecordID` type out of the waiter.

Reply:
This still says `RequestID`. Have I missed something?

Reply:
It used to be called `ID` and was on the type `Waiter`, but dependent on the `HTTPScheduledRequest`

Reply:
I see, I was confused by this comment saying the change was to `RecordID`.

Reply:
https://github.com/swift-server/async-http-client/pull/397#discussion_r666253889